internal/v2/controller/http/command.go

```diff
@@ -0,0 +1,133 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/cache"
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/internal/v2/application"
+	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/gorilla/mux"
+)
+
+const SDKPostEventReserved = "ds-postevent"
+const SDKReturnEventReserved = "ds-returnevent"
+
+func (c *V2HttpController) Command(writer http.ResponseWriter, request *http.Request) {
+	var reserved url.Values
+	vars := mux.Vars(request)
+	ds := container.DeviceServiceFrom(c.dic.Get)
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := checkServiceLocked(request, ds.AdminState)
```

Comment:
it's in the v2 controller (and yeah the request parameter isn't necessary here):
https://github.com/edgexfoundry/device-sdk-go/blob/833c05b782f270c396ee40879762f076aaf9331e/internal/v2/controller/http/controller.go#L75-L81

I'm wondering do we even need to functionize this since it's logic is so straightforward
`if err := checkServiceLocked(); err != nil` vs `if Adminstate == contract.Locked`, there's some place we directly use the if statement for example:
https://github.com/hahattan/device-sdk-go/blob/14a705dd23cc4d0b9f21bc0cf06a165c8906fa41/internal/v2/controller/http/command.go#L92-L95

Reply:
Sure, but this code should be in the app layer since it is needed for any protocol. Getting the device service from container also. Make it first thing CommandHandler does?